homeassistant/components/thread/discovery.py

```diff
@@ -121,7 +121,19 @@ def async_read_zeroconf_cache(aiozc: AsyncZeroconf) -> list[ThreadRouterDiscover
             # data is not fully in the cache, so ignore for now
             continue
 
-        results.append(async_discovery_data_from_service(info))
+        # Service properties are always bytes if they are set from the network.
+        # For legacy backwards compatibility zeroconf allows properties to be set
+        # as strings but we never do that so we can safely cast here.
+        service_properties = cast(dict[bytes, bytes | None], info.properties)
+
+        if not (xa := service_properties.get(b"xa")):
+            _LOGGER.debug("Ignoring record without xa %s", info)
+            continue
```

Comment:
It seems that old border routers didn't had that property set (2021 and earlier). In other words, this could actually happen in practice. But we are not interested in such old border routers, since they likely are not interesting for the protocols we are aiming for (Matter). So ignoring is the right approach.

I think I'd use a higher log level though, maybe info or even warning.

Reply:
FWIW, this is for diagnostic only. The log level is adjusted below in `_add_update_service` :+1: